test/Datadog.Trace.ClrProfiler.IntegrationTests/AWS/AwsSqsTests.cs

```diff
@@ -0,0 +1,169 @@
+            AwsSqsSpan.GetDefault("ReceiveMessage"),
+            AwsSqsSpan.GetDefault("DeleteMessageBatch"),
+
+            // SendMessagesWithoutInjectedHeadersAsync
+            AwsSqsSpan.GetDefault("SendMessage"),
+            AwsSqsSpan.GetDefault("ReceiveMessage"),
+            AwsSqsSpan.GetDefault("DeleteMessage"),
+            AwsSqsSpan.GetDefault("SendMessageBatch"),
+            AwsSqsSpan.GetDefault("ReceiveMessage"),
+            AwsSqsSpan.GetDefault("DeleteMessageBatch"),
+
+            // SendMessageAsync
+            AwsSqsSpan.GetDefault("SendMessage"),
+            AwsSqsSpan.GetDefault("SendMessage"),
+
+            // ReceiveMessageAndDeleteMessageAsync
+            AwsSqsSpan.GetDefault("ReceiveMessage"),
+            AwsSqsSpan.GetDefault("DeleteMessage"),
+            AwsSqsSpan.GetDefault("ReceiveMessage"),
+            AwsSqsSpan.GetDefault("DeleteMessage"),
+
+            // SendMessageBatchAsync
+            AwsSqsSpan.GetDefault("SendMessageBatch"),
+            AwsSqsSpan.GetDefault("SendMessageBatch"),
+
+            // ReceiveMessagesAndDeleteMessageBatchAsync
+            AwsSqsSpan.GetDefault("ReceiveMessage"),
+            AwsSqsSpan.GetDefault("DeleteMessageBatch"),
+            AwsSqsSpan.GetDefault("ReceiveMessage"),
+            AwsSqsSpan.GetDefault("DeleteMessageBatch"),
+
+            // DeleteQueueAsync
+            AwsSqsSpan.GetDefault("DeleteQueue"),
+            AwsSqsSpan.GetDefault("DeleteQueue"),
+        };
+
+        public AwsSqsTests(ITestOutputHelper output)
+            : base("Aws.Sqs", output)
+        {
+        }
+
+        [Theory]
+        [MemberData(nameof(PackageVersions.AwsSqs), MemberType = typeof(PackageVersions))]
+        [Trait("Category", "EndToEnd")]
+        public void SubmitsTraces(string packageVersion)
+        {
+            SetCallTargetSettings(enableCallTarget: true, enableMethodInlining: true);
+
+            int agentPort = TcpPortProvider.GetOpenPort();
+            using (var agent = new MockTracerAgent(agentPort))
+            using (var processResult = RunSampleAndWaitForExit(agent.Port, packageVersion: packageVersion))
+            {
+                Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+
+                var spans = agent.WaitForSpans(_expectedSpans.Count, operationName: AwsExpectation.IntegrationOperationName);
+                spans.Should().HaveCountGreaterOrEqualTo(_expectedSpans.Count);
+
+                spans.OrderBy(s => s.Start).Should().BeEquivalentTo(_expectedSpans, options => options
+                    .WithStrictOrdering()
+                    .ExcludingMissingMembers()
+                    .ExcludingDefaultSpanProperties()
+                    .AssertMetricsMatchExcludingKeys("_dd.tracer_kr", "_sampling_priority_v1")
+                    .AssertTagsMatchAndSpecifiedTagsPresent("env", "aws.requestId", "aws.queue.url"));
```

Comment:
Very slick! 🎉 Though feels like trying to create our own version of Verify 😄

Reply:
Thanks! I think I started this branch before we had started testing Verify so I thought it would be a great way to express the key tags and metrics. I don't really want to scratch this because it's doing its job and we haven't merged Verify yet..